src/Microsoft.Framework.Configuration/ConfigurationSection.cs

```diff
@@ -30,6 +36,7 @@ public ConfigurationSection(IList<IConfigurationSource> sources, string key)
             }
 
             _key = key;
+            _prefix = prefix;
```

Comment:
Assuming we can disallow empty `prefix`, it seems we won't be using the `prefix` for anything but to calculate `Path`. On the other hand we use `Path` in a number of places. 
Wondering if we should just get rid of the `_prefix` field, and move the assignment of a `_path` field here to the constructor.